src/tests/shamalgs/algorithm/algorithmTests.cpp

```
@@ -8,18 +8,46 @@
-    test.check();
+
+    if (!shamalgs::primitives::impl::is_impl_set_sort_by_key_pow2_len()) {
+        shamalgs::primitives::impl::autoselect_impl_sort_by_key_pow2_len();
+    }
+    auto current_impl = shamalgs::primitives::impl::get_current_impl_sort_by_key_pow2_len();
+
+    for (const std::string &impl :
+         shamalgs::primitives::impl::get_default_impl_list_sort_by_key_pow2_len()) {
+        shamalgs::primitives::impl::set_impl_sort_by_key_pow2_len(impl);
+        shamlog_info_ln("tests", "testing implementation:", impl);
+        test.check();
```

Cppcheck (2.21.0)

[style] The function 'add_data' is never used.

(unusedFunction)